pregenerate_exercises.py

```diff
@@ -0,0 +1,284 @@
+import argparse
+import json
+import os
+import re
+import shutil
+import subprocess
+import yaml
```

Comment:
yaml module does not necessarily exist in the system by default. The instructions should say that e.g. "pip3 install pyyaml" is needed.